src/PropertySuggester/Suggesters/SimpleSuggester.php

```diff
@@ -73,38 +78,60 @@ public function setInitialSuggestions( array $initialSuggestionIds ) {
 	}
 
 	/**
-	 * @param int[] $propertyIds
-	 * @param string[] $idTuples
+	 * @param float $classifyingConditionWeight
+	 */
+	public function setClassifyingConditionWeight( $classifyingConditionWeight ) {
+		$this->classifyingConditionWeight = $classifyingConditionWeight;
+	}
+
+	/**
+	 * @param array[] $idTuples array of ( int Property id, int Item id) tuples
 	 * @param int $limit
 	 * @param float $minProbability
 	 * @param string $context
+	 *
 	 * @throws InvalidArgumentException
+	 *
 	 * @return Suggestion[]
 	 */
-	protected function getSuggestions( array $propertyIds, array $idTuples, $limit, $minProbability, $context ) {
```

Comment:
It looks like the removal of this parameter could easily be a separate patch. Splitting this into more smaller patches would help a lot in getting reviews for this significant change. Same for the removal of `protected`. I'm willing to help and upload such patches, if there is a chance they get merged fast.